.github/workflows/ci.yml

```diff
@@ -745,3 +748,126 @@ jobs:
           python3 -m pytest -s ./graphscope/tests/kubernetes/test_demo_script.py -k test_demo_on_hdfs
           # Check the result file have successfully written to the given location
           # hdfs dfs -test -e /ldbc_sample/res.csv_0 && hdfs dfs -test -e /ldbc_sample/res.csv_1
+
+  spark-graphx-test:
+    runs-on: [self-hosted, ubuntu2004]
+    if: ${{ github.repository == 'alibaba/GraphScope' }}
+    needs: [build-wheels]
+    steps:
+      - uses: actions/checkout@v2.3.2
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+
+      - name: Add envs to GITHUB_ENV
+        run: |
+          short_sha=$(git rev-parse --short HEAD)
+          echo "SHORT_SHA=${short_sha}" >> $GITHUB_ENV
+
+      - name: Prepare Environment
+        env:
+          GS_TEST_DIR: ${{ github.workspace }}/gstest
+        run: |
+          # download dataset
+          git clone -b master --single-branch --depth=1 https://github.com/7br/gstest.git ${GS_TEST_DIR}
+
+      - name: Setup SSH
```

Comment:
why do we need these, are they essential?

Reply:
The purpose is to test an end-to-end case for GS & GraphX integration, and this job will only be triggered when graphx-related code is modified.